lib/fine-types/src/Language/FineTypes/Export/OpenAPI/Value/ToJSON.hs

```diff
@@ -84,20 +99,7 @@ jsonFromValue = go
     go1 Typ.Option _ (Option Nothing) = JS.object []
     go1 Typ.Sequence t (Sequence xs) = JS.toJSON $ map (go t) xs
     go1 Typ.PowerSet t (PowerSet xs) = JS.toJSON $ map (go t) $ Set.toList xs
-    go1 _ _ _ = error "jsonFromValue: Typ error"
-
-    go2 :: TwoF Value Value -> JS.Value
-    go2 (FiniteMap _) =
-        error "FiniteMapV is not supported by JSON"
-
--- | Flatten a chain of @n@ binary products to a single 'Product'.
-flattenBinaryProduct :: Int -> Value -> [Value]
-flattenBinaryProduct = flatten
-  where
-    flatten :: Int -> Value -> [Value]
-    flatten 1 x = [x]
-    flatten n (Product [x, y]) = x : flatten (n - 1) y
-    flatten _ x = [x]
+    go1 t _ v = impossibleCase t v "Impossible case"
```

Comment:
```suggestion
    go1 t _ v = impossibleCase t v "Value does not type check"
```

Likewise — the case is impossible because it violates the assumption that the value typechecks.